peekingduck/pipeline/nodes/model/jde_mot/jde_model.py

```diff
@@ -0,0 +1,182 @@
+# Modifications copyright 2021 AI Singapore
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#      https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Original copyright (c) 2019 ZhongdaoWang
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+JDE model for human detection and tracking
+"""
+
+from typing import Any, Dict, List, Tuple
+import logging
+import numpy as np
+import cv2
+import torch
+from peekingduck.pipeline.nodes.model.jde_mot.jde_files.utils.utils import letterbox
+from peekingduck.pipeline.nodes.model.jde_mot.jde_files.utils.parse_config import (
+    parse_model_cfg,
+)
+from peekingduck.pipeline.nodes.model.jde_mot.jde_files.multitracker import JDETracker
+from peekingduck.weights_utils import checker, downloader, finder
+
+# pylint: disable=invalid-name, no-member, no-self-use, attribute-defined-outside-init, logging-fstring-interpolation, too-few-public-methods
+
+
+class JDE:
```

Comment:
Refactor to `JDEModel` instead to be consistent with other models in PKD? 
https://github.com/aimakerspace/PeekingDuck/blob/dev/peekingduck/pipeline/nodes/model/posenetv1/posenet_model.py#L25
https://github.com/aimakerspace/PeekingDuck/blob/dev/peekingduck/pipeline/nodes/model/yolov4/yolo_model.py#L28

Reply:
Refactored